File: Python/moneyManagement.py
solde = 200000.0
historique_transactions = []

def is_numeric(value):
    return value.replace(".", "", 1).isdigit()

def is_positive_numeric(value):
    return is_numeric(value) and float(value) > 0

def manage_money():
    global solde  # Declare solde as a global variable
    global historique_transactions
    
    while True:
        sub_menu_1 = int(input("1. Versement 2. Retrait 3. Consultation du solde 4. Historique des transactions 5. Retour Choisissez une option : "))

        if sub_menu_1 not in [1, 2, 3, 4, 5]:
            print("Veuillez saisir une option valable")
            continue

        if sub_menu_1 == 1:
            versement = input("Combien souhaitez-vous verser : ")
            if is_positive_numeric(versement):
                solde += float(versement)
                historique_transactions.append(f"Versement de {versement}")
                print("Versement effectué avec succès !")
            else:
                print("Veuillez entrer un montant valide.")
        elif sub_menu_1 == 2:
            retrait = input("Combien souhaitez-vous retirer : ")
            if is_positive_numeric(retrait):
                if float(retrait) > solde:
                    print("Désolé, votre solde est insuffisant pour ce retrait")
                else:
                    solde -= float(retrait)
                    historique_transactions.append(f"Retrait de {retrait}")
                    print("Retrait effectué avec succès !")
            else:
                print("Veuillez entrer un montant valide.")
        elif sub_menu_1 == 3:
            print(f"Votre solde est : {solde}")
        elif sub_menu_1 == 4:
            print("Historique des transactions :")
            for transaction in historique_transactions:
                print(transaction)
        elif sub_menu_1 == 5:
            print("Retour au menu principal.")
            break

File: JS/moneyManagement.js
const prompt = require("prompt-sync")();
let solde = 200000;
let historiqueTransactions = [];

function isNumeric(value) {
    return !isNaN(parseFloat(value)) && isFinite(value);
}

function isPositiveNumeric(value) {
    return isNumeric(value) && parseFloat(value) > 0;
}

function manageMoney() {
    while (true) {
        let subMenu1 = +prompt("1. Versement 2. Retrait 3. Consultation du solde 4. Historique des transactions 5. Retour Choisissez une option : ");

        if (![1, 2, 3, 4, 5].includes(subMenu1)) {
            console.log("Veuillez saisir une option valable");
            continue;
        }

        if (subMenu1 === 1) {
            let versement = +prompt("Combien souhaitez-vous verser : ");
            if (isPositiveNumeric(versement)) {
                solde += versement;
                historiqueTransactions.push(`Versement de ${versement}`);
                console.log("Versement effectué avec succès !");
            } else {
                console.log("Veuillez entrer un montant valide.");
            }
        }
        else if (subMenu1 === 2) {
            let retrait = +prompt("Combien souhaitez-vous retirer : ");
            if (isPositiveNumeric(retrait)) {
                if (retrait > solde) {
                    console.log("Désolé, votre solde est insuffisant pour ce retrait");
                } else {
                    solde -= retrait;
                    historiqueTransactions.push(`Retrait de ${retrait}`);
                    console.log("Retrait effectué avec succès !");
                }
            } else {
                console.log("Veuillez entrer un montant valide.");
            }
        }
        else if (subMenu1 === 3) {
            console.log(`Votre solde est : ${solde}`);
        }
        else if (subMenu1 === 4) {
            console.log("Historique des transactions :");
            for (let transaction of historiqueTransactions) {
                console.log(transaction);
            }
        }
        else if (subMenu1 === 5) {
            console.log("Retour au menu principal.");
            break; 
        }
    }
}


module.exports = {
    manageMoney
};